vivisect/analysis/generic/symswitchcase.py

```diff
@@ -0,0 +1,1293 @@
+            #        unks, 
+            #        '\n'.join([repr(x) for x in cspath[1]]), 
+            #        '\n'.join([repr(x) for x in aspath[1]]))
+            raise SymIdxNotFoundException(cspath, aspath)
+            
+        return potentials[0]
+
+    def getCountOffset(self):
+        '''
+        Limited value function.
+        '''
+        lower, upper, offset = self.getBounds()
+        if None in (lower, upper): 
+            logger.info("something odd in count/offset calculation... skipping 0x%x...", self.jmpva)
+            return
+        return (upper-lower+1), offset+lower
+
+
+    def getConstraints(self):
+        csp, asp, fullp = self.getSymbolikParts()
+
+        cons = [eff for eff in fullp[1] if eff.efftype==EFFTYPE_CONSTRAIN]
+        [con.reduce() for con in cons]
+        return cons
+
+    def getSymbolikParts(self, next=False):
+        '''
+        Puts together a path from the start of the function to the jmpva, breaking off the last 
+        codeblock, and returning the symbolik path for both sections:
+        cspath = (emu, effs) for the "context" path (from function start)
+        aspath = (emu, effs) for the last codeblock
+
+        if "next" isn't False, the next iteration is returned.  otherwise, the current iteration is (same as last call).
+        this allows the Symbolik Parts to be easily accessible from various parts of the algorithm without handing around
+        a lot of state as function args.
+        '''
+        if not next and self.cspath is not None and self.aspath is not None and self.fullpath is not None:
+            return self.cspath, self.aspath, self.fullpath
+        
+        self.clearCache()
+
+        vw = self.vw
+        sctx = self.sctx
+        jmpva = self.jmpva
+
+        fva = vw.getFunction(jmpva)
+        cb = vw.getCodeBlock(jmpva)
+        if cb is None:
+            raise Exception("Dynamic Branch is not currently part of a CodeBlock!")
+        cbva, cbsz, cbfva = cb
+
+        if self._sgraph is None:
+            self._sgraph = sctx.getSymbolikGraph(fva)
+
+        if self._codepathgen is None:
+            #self._codepathgen = viv_graph.getCodePathsTo(self._sgraph, cbva)
+            pathGenFactory = viv_graph.PathGenerator(self._sgraph)
+            self._codepathgen = pathGenFactory.getFuncCbRoutedPaths(fva, cbva, 1, timeout=20)
+
+        self._codepath = self._codepathgen.next()
+        contextpath = self._codepath[:-1]
+        analpath = self._codepath[-1:]
+
+        self.cspath = sctx.getSymbolikPaths(fva, graph=self._sgraph, args=[], paths=[contextpath]).next()
+        self.aspath = sctx.getSymbolikPaths(fva, graph=self._sgraph, args=[], paths=[analpath]).next()
+        if len(self.cspath[1]) == 0:
+            self.cspath = self.aspath
+        self.fullpath = sctx.getSymbolikPaths(fva, graph=self._sgraph, args=[], paths=[self._codepath]).next()
+
+
+        return self.cspath, self.aspath, self.fullpath
```

Comment:
So as a structure question, I'm curious why you chose not to go with getSymbolikPathsTo from the SymbolikAnalysisContext class?

Reply:
there's some magic here that getSymbolikPathsTo doesn't allow, namely the matching up of two different symboliks extrusions with different "starting points"